dbt_subprojects/tokens/models/transfers_and_balances/gnosis/tokens_gnosis_suicide_transfers.sql

```diff
@@ -0,0 +1,134 @@
+{{config(
+    schema = 'tokens_gnosis',
+    alias = 'suicide_transfers',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+    unique_key = ['block_date','unique_key'],
+)
+}}
+
+WITH 
+
+suicide AS (
+    SELECT 
+         cast(date_trunc('day', block_time) as date) AS block_date
+        , block_time
+        , block_number
+        , tx_hash
+        , tx_index
+        , trace_address
+        , tx_from
+        , tx_to
+        , address 
+        , refund_address 
+    FROM 
+        {{ source('gnosis', 'traces') }}
+    WHERE
+        type = 'suicide'
+        AND
+        success
```

Comment:
are you intentionally leaving out incremental filters on this one?

Reply:
This was intentional, mostly because it feels the implementation of the incremental is non-trivial and it would be hard for me to test it. Some more details:

The `tokens_gnosis_suicide_transfers.sql` is a particular model. The whole point of the `transfers` model is to have a table that will allow to build the balances. However, we need to know the balances of the contracts that are destroyed in order to build this table.

So the idea is that every time a contract is destroyed one can refund the native currency that the contract has into a refund address. So one needs to know the balance, which means we need the cumulative sum of all transfers (in/out) of this contract up to the point of destruction. 

I see two pain points on the incremental implementation:
1. This would mean that the incremental should keep track of the real balance block by block, and not loose this info when moving to an new partition. Not that difficult, but needs to be tested that we do not "leak" balances when changing partitions during incremental updates; 
2. We do not know which contract will be destroyed, so we would need to keep the balance of all contracts even though we are just trying to build the transfers table.

So, I am happy is there is an incremental implementation for this model. But I think it can not be the standard incremental filters used for the other models.

Reply:
all good, the explanation as to why is all i needed 🤝 

the performance on this spell looks pretty good as-is, taking only 33s to build full history